api/app/models/comment.rb

```diff
@@ -3,19 +3,17 @@ class Comment < ApplicationRecord
 
   belongs_to :user
   belongs_to :parent, polymorphic: true
+  belongs_to :issue, counter_cache: true
   has_many :replies, class_name: 'Comment', as: :parent, dependent: :destroy
 
   validates :content, presence: true
+  before_validation :set_issue
 
   scope :oldest_first, -> { order(created_at: :asc) }
 
-  def comments_count
-    total = 1
+  private
 
-    replies.each do |reply|
-      total += reply.comments_count
-    end
-
-    total
+  def set_issue
+    self.issue = parent.is_a?(Comment) ? parent.issue : parent
   end
```

Comment:
I'm changing a bit the approach for calculating the amount of comments and upvotes. Currently both are just methods that query the db to count on SQL the amount of records o each kind associated to the issue.
For comments it's specially complicated because of the polymorphic association (which is also a self join), so the method really counts the child comments associated to each comment and adds them all up in a recursive call. If I add on top of that the requirement to sort on those values, it would be a mess. I decided to move to a simpler approach and use counter caches: Basically there will be an `upvotes_count` and a `comments_count` column in the `issues` table which will store the amount of records associated to each issue. Each column will get updated every time there is a new comment or a new upvote, or every time one of them is deleted.

ActiveRecord is handling these counts updates automatically https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache

Reply:
cool

<img width="676" alt="image" src="https://github.com/mapra99/feedbackr/assets/26731448/7b17d32c-e471-4c88-8096-20c8be1c345c">